ballerina/stream_implementer.bal

```diff
@@ -57,9 +62,20 @@ class SOQLQueryResultStream {
         self.nextRecordsUrl = response.hasKey(NEXT_RECORDS_URL) ? check response.get(NEXT_RECORDS_URL).ensureType() :
             EMPTY_STRING;
 
-        record{}[] array = response.records;
-        return array;
+        record {}[]|error returnData = self.recordConverter(response.records, self.returnType);
```

Comment:
Do we need to jump to java side for this? Cant we use cloneWithtype()?

Reply:
Can we pass a typedesc to cloneWithtype()? ok i'll add that her ethen